### PR TITLE
Feat: sign protocol parameters and epoch in certificate

### PR DESCRIPTION
**Pull request by jpraynaud**
## Content

This PR includes a **new computation of the protocol message** in the `Pythagoras` Mithril era:
- The **next protocol parameters** are appended to the protocol message
- The **current epoch** is appended to the protocol message

### Version Updates:

> [!IMPORTANT]
> :fire: This PR introduces a breaking-change in the `mithril-client` and `mithril-client-wasm` crates.

* [`examples/client-mithril-stake-distribution/Cargo.toml`](diffhunk://#diff-887fc60a9b9bc268630f88ffb4746333e0f8c76d0a2dc8800abe456fd8c2ce39L4-R4): Updated the version from `0.1.17` to `0.2.0`.
* [`mithril-aggregator/Cargo.toml`](diffhunk://#diff-5ae504281312c194e5e8cf6aa890f49508756041db9dbce0fd89fbf04438f90dL3-R3): Updated the version from `0.5.68` to `0.5.69`.
* [`mithril-client-cli/Cargo.toml`](diffhunk://#diff-2a39f2159ddf8206fa697498383f3b5c626c184d3da53ac7417e4fb1b22351f0L3-R3): Updated the version from `0.9.12` to `0.9.13`.
* [`mithril-client-wasm/Cargo.toml`](diffhunk://#diff-72ad6e1c1356d0f73e50d9e7970bd2defe29e80296db55ad0c0ad44362a6ea84L3-R3): Updated the version from `0.4.1` to `0.5.0`.
* [`mithril-client/Cargo.toml`](diffhunk://#diff-a272fa86be295b0d30ea31b0ca6a5f7b915a4f52677bd575843152082b15a4beL3-R3): Updated the version from `0.8.18` to `0.9.0`.
* [`mithril-common/Cargo.toml`](diffhunk://#diff-0b137373460b5fa18b029b657531df061019d3ce8efadc17b522b1a104971d65L3-R3): Updated the version from `0.4.59` to `0.4.60`.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

### CODE REVIEW

**Review by sfauvel — APPROVED** [review]
LGTM
just some suggestions on tests

**Review by Alenar — APPROVED** [review]

**Review by dlachaume — APPROVED** [review]
LGTM 🚀, just a minor suggestion.

File: mithril-signer/src/services/signable_builder/signable_seed_builder.rs [hunk 1]
```diff
@@ -128,4 +179,59 @@ mod tests {
...
+    #[tokio::test]
+    async fn test_compute_next_protocol_parameters_protocol_message_value() {
+        let epoch = Epoch(5);
+        let next_fixture = MithrilFixtureBuilder::default().with_signers(4).build();
+        let protocol_initializer = next_fixture.signers_fixture()[0]
+            .protocol_initializer
+            .clone();
+        let expected_next_protocol_parameters =
+            Into::<ProtocolParameters>::into(protocol_initializer.get_protocol_parameters())
+                .compute_hash();
```
[the remark this answers is not in the document]
Comment (jpraynaud) [reply mithril-signer/src/services/signable_builder/signable_seed_builder.rs:186]: I modified the test to make it more readable

File: mithril-aggregator/src/services/signable_builder/signable_seed_builder.rs [hunk 2]
```diff
@@ -64,10 +78,20 @@ mod tests {
...
         )));
-        let signable_seed_builder = AggregatorSignableSeedBuilder::new(epoch_service);
+
+        AggregatorSignableSeedBuilder::new(epoch_service)
+    }
+
+    #[tokio::test]
+    async fn test_compute_next_aggregate_verification_key_protocol_message_value() {
+        let epoch = Epoch(5);
+        let fixture = MithrilFixtureBuilder::default().with_signers(5).build();
```
Comment (curiecrypt) [inline mithril-aggregator/src/services/signable_builder/signable_seed_builder.rs:88]: num of signers

File: mithril-aggregator/src/services/signable_builder/signable_seed_builder.rs [hunk 3]
```diff
@@ -64,10 +78,20 @@ mod tests {
...
+        let next_fixture = MithrilFixtureBuilder::default().with_signers(4).build();
+        let signable_seed_builder = build_signable_builder_service(epoch, &fixture, &next_fixture);
+        let expected_next_aggregate_verification_key = next_fixture.compute_and_encode_avk();
 
         let next_aggregate_verification_key = signable_seed_builder
-            .compute_next_aggregate_verification_key_protocol_message_part_value()
+            .compute_next_aggregate_verification_key()
```
Comment (curiecrypt) [inline mithril-aggregator/src/services/signable_builder/signable_seed_builder.rs:94]: Difference?

File: mithril-client-cli/src/commands/mithril_stake_distribution/download.rs [hunk 4]
```diff
@@ -113,7 +113,7 @@ impl MithrilStakeDistributionDownloadCommand {
             "Verify that the Mithril stake distribution is signed in the associated certificate",
         )?;
         let message = MessageBuilder::new()
-            .compute_mithril_stake_distribution_message(&mithril_stake_distribution)
+            .compute_mithril_stake_distribution_message(&certificate, &mithril_stake_distribution)
             .with_context(|| {
                 "Can not compute the message for the given Mithril stake distribution"
```
Comment (curiecrypt) [inline mithril-client-cli/src/commands/mithril_stake_distribution/download.rs:118]: maybe? change the error message?

**Review by curiecrypt — COMMENTED** [review]
LGTM 👍

**Pull request merged**